osltweaks.php

```diff
@@ -44,6 +44,29 @@ function osltweaks_civicrm_buildForm($formName, &$form) {
         }
       }
     }
+
+    // If contribution page id exist in $settings['member_redirect_contribution_pages'] array
+    if (array_key_exists($form->_id, ($settings['member_redirect_contribution_pages'] ?? array()))) {
+      // Get current contact id
+      $currentContactId = CRM_Core_Session::singleton()->getLoggedInContactID();
+        // Get memberships base on the current contact id
+      $memberships = civicrm_api3('Membership', 'get', [
```

Comment:
@lucky091588 because we only care if they have any membership record, regardless of status, you can just use the membership.getCount api here, and if the count is more than 0, then redirect. This way you don't need the foreach and don't need to check status. Make sense?